### PR TITLE
Set Up Community Endpoint

### PR DESCRIPTION
**Pull request by AdrianAndersen**
This PR adds community functionality to add/remove/accept/deny friends. Also adds push notifs for friend requests and filters out sessions and configs so you only see yours and your friends'

Related to https://github.com/mathiazom/rezervo-web/pull/99

### CODE REVIEW

File: rezervo/api/community.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+from rezervo.api.common import get_db, token_auth_scheme
+from rezervo.database import crud
+from rezervo.schemas.community import Community
+from rezervo.settings import Settings, get_settings
+
+router = APIRouter()
+
+
+@router.get("/community", response_model=Community)
+def get_users(
```
Comment (mathiazom) [inline rezervo/api/community.py:14]:
```suggestion
def get_community(
```

Comment (AdrianAndersen) [reply rezervo/api/community.py]: Ah forgot to change that. ✅

File: rezervo/database/crud.py [hunk 2]
```diff
@@ -372,3 +373,31 @@ def purge_slack_receipts(db) -> int:
...
+        .order_by(models.User.name)
+        .all()
+    )
+    chain_users = (
+        db.query(models.ChainUser).filter(models.ChainUser.user_id != user_id).all()
+    )
+
+    user_to_chain_map: Dict[UUID, List[str]] = {}
+    for chain_user in chain_users:
+        user_to_chain_map.setdefault(chain_user.user_id, []).append(chain_user.chain)
```
Comment (mathiazom) [inline rezervo/database/crud.py:391]: Have you considered using [`defaultdict`](https://docs.python.org/3/library/collections.html#collections.defaultdict)?

Comment (AdrianAndersen) [reply rezervo/database/crud.py]: (https://github.com/mathiazom/rezervo/pull/33/commits/9ff1777e9bf573505933d3c66796c7054cc9b942)

Comment (AdrianAndersen) [reply rezervo/database/crud.py]: Smart, did not know that was a thing

File: rezervo/api/community.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    settings: Settings = Depends(get_settings),
+):
+    db_user = crud.user_from_token(db, settings, token)
+    if db_user is None:
+        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED)
+    users = crud.get_community(db, db_user.id)
+    return users
+
+
+@router.put("/community/update-relationship", response_model=UserRelationship)
```
Comment (mathiazom) [inline rezervo/api/community.py:32]: Implicit update when using `PUT` right?
```suggestion
@router.put("/community/relationship", response_model=UserRelationship)
```

File: rezervo/api/community.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    settings: Settings = Depends(get_settings),
+):
+    db_user = crud.user_from_token(db, settings, token)
+    if db_user is None:
+        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED)
+    updated_relationship = crud.modify_user_relationship(
+        db, db_user.id, payload.user_id, payload.action
+    )
+
+    if updated_relationship is UserRelationship.REQUEST_SENT:
```
Comment (mathiazom) [inline rezervo/api/community.py:46]: Have you considered moving the notification logic to `modify_user_relationship`?

File: rezervo/api/community.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+def get_community(
+    token=Depends(token_auth_scheme),
+    db: Session = Depends(get_db),
+    settings: Settings = Depends(get_settings),
+):
+    db_user = crud.user_from_token(db, settings, token)
+    if db_user is None:
+        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED)
+    users = crud.get_community(db, db_user.id)
+    return users
```
Comment (mathiazom) [inline rezervo/api/community.py:29]:
```suggestion
    return crud.get_community(db, db_user.id)
```

File: rezervo/api/community.py (hunk 4 above)
Comment (AdrianAndersen) [reply rezervo/api/community.py:46]: Yes, but the `push.py` module has already has imports from `crud.py`. Having this code in `crud.py` results in a circular import

File: rezervo/api/community.py (hunk 5 above)
Comment (AdrianAndersen) [reply rezervo/api/community.py]: Fixed in https://github.com/mathiazom/rezervo/pull/33/commits/46d54be4eca813e0bb0e92e3fa1edca4403679cd

File: rezervo/api/community.py (hunk 4 above)
Comment (mathiazom) [reply rezervo/api/community.py:45]: Ah right :upside_down_face:  It might be a bad idea to have a push side effect in crud anyway, so this is fine :+1:

**Review by mathiazom — APPROVED** [review]

**Pull request merged**